scripts/w3c.js

```diff
@@ -1,354 +1,647 @@
+    for (const [name, alias] of Object.entries(aliasesToInvert)) {
+        // Look for the previously targeted entry.
+        // If it does not exist, that's fine, we'll just ignore it.
+        // Tests will catch any hiccup we may create in any case.
+        const match = name.match(/^(.+)-([^-]+)/);
+        if (!match) {
+            continue;
+        }
+        entry = current[match[1]];
+        if (!entry || !entry.versions) {
+            continue;
+        }
+        if (entry.versions[match[2]]) {
+            console.log(`- ${match[1]} (${match[2]}): make alias of ${alias}`);
+            entry.versions[match[2]] = { aliasOf: alias };
+        }
     }
-}
 
-function walk(obj) {
-    for (var i=1; i < arguments.length; i++) {
-        var prop = arguments[i]
-        if (prop in obj) {
-            obj = obj[prop];
-        } else {
-            return void 0;
+    console.log("Sorting references...");
+    const sorted = helper.sortRefs(current);
+
+    console.log("Updating obsoletes properties...");
+    const superseders = {};
+    for (const [shortname, entry] of Object.entries(current)) {
+        if (entry.obsoletes) {
+            delete entry.obsoletes;
         }
+        for (const superseder of (entry.obsoletedBy ?? [])) {
+            if (!superseders[superseder]) {
+                superseders[superseder] = [];
+            }
+            superseders[superseder].push(shortname);
+        }
+    }
+    for (const [shortname, obsoletes] of Object.entries(superseders)) {
+        const entry = current[shortname];
+        if (!entry) {
+            continue;
+        }
+        obsoletes.sort();
+        entry.obsoletes = obsoletes;
     }
-    return obj;
+
+    console.log("Writing file...");
+    helper.writeBiblio(FILENAME, sorted);
+    helper.tryOverwrite(FILENAME);
 }
 
-function _cloneJSON(obj) {
-    return JSON.parse(JSON.stringify(obj));
+if (!since) {
+    // Compute the most recent W3C spec date in Specref, and substract a
+    // few weeks to get a threshold date beyond which we consider that the
+    // data in Specref no longer needs to be updated.
+    const latestUpdate = Object.values(current).reduce((version, curr) =>
+        curr.rawDate > version ? curr.rawDate : version, '');
+    const twoWeeks = 2 * 7 * 24 * 60 * 60 * 1000;
+    const thresholdDate = new Date(latestUpdate);
+    thresholdDate.setTime(thresholdDate.getTime() - twoWeeks);
+    since = thresholdDate
+        .toISOString()
+        .substring(0, 10);
 }
 
-function isGeneratedByThisScript(ref) {
-    return ref.source == "https://www.w3.org/2002/01/tr-automation/tr.rdf" || ref.source == "http://www.w3.org/2002/01/tr-automation/tr.rdf" || ref.source == RDF_FILE;
+const start = Date.now();
+if (start - (new Date(since)).getTime() > 365 * 24 * 3600 * 1000) {
+    // A full update can take more than an hour. Let's alert the user
+    // (text on a red background).
```

Comment:
```suggestion
if (start - (new Date(since)).getTime() > 365 * 24 * 3600 * 1000) {
    // Update from 1+ year old specifications can take more than an hour
    // Let's alert the user (text on a red background).
```